slsim/Sources/SourceTypes/point_source.py

```diff
@@ -0,0 +1,105 @@
+from slsim.Sources.SourceTypes.supernova_event import SupernovaEvent
+from slsim.Sources.SourceTypes.quasar import Quasar
+from slsim.Sources.SourceTypes.general_lightcurve import GeneralLightCurve
+
+_SUPPORTED_POINT_SOURCES = ["supernova", "quasar", "general_lightcurve"]
+
+
+class PointSource(object):
+    """Class to manage a single point source."""
+
+    def __init__(self, source_dict, cosmo=None, **kwargs):
```

Comment:
make point_source_type and perhaps all the parts that are explicitly being managed by this class explicit inputs to this class